src/loki/Counters.h

```diff
@@ -0,0 +1,53 @@
+/* Copyright (C) 2019 European Spallation Source, ERIC. See LICENSE file */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Holds application counters
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <cinttypes>
+
+struct Counters {
+  // Input Counters - accessed in input thread
+  int64_t RxPackets;
+  int64_t RxBytes;
+  int64_t FifoPushErrors;
+  int64_t InputIdle;
+  int64_t PaddingFor64ByteAlignment[4]; // cppcheck-suppress unusedStructMember
```

Comment:
is this necessary when the "aligned" attribute is also present?

Reply:
Yes. Aligning only ensures that the struct starts at a cache line but the uint64_t are only 8 bytes. So to ensure that input and processing threads do not write to the same cache line we need the padding. Not sure that this is critical for performance though but I thought it a good practice.

Reply:
does the Input and Processing thread both write to the same struct? If that's the same I think we should either have a struct per thread or atomic ints in the stats. Is there any other threading relationship, like control logic, other than plain counters?